how-to/what-is-vm-templating-and-how-do-I-use-it.md

```diff
@@ -0,0 +1,54 @@
+# What Is VM Template and How To Enable It
+
+### What is VM template
+VM template is a Kata containers feature that enables new VMs
+creation using cloning technique. When enabled, new VMs are created
+by cloning from a pre-created template VM, and they will share the
+same initramfs, kernel and agent memory in readonly mode. It is very
+much like a process fork done by the kernel but here we *fork* VMs.
+
+### What are the Pros
+VM template helps speeding up new container creation and saves a lot
+of memory if there are many kata containers running on the same host.
+If you are running a density workload, or care a lot about container
+startup speed, VM template can be very useful.
+
+In one example, we created 100 Kata containers each claiming 128MB
+guest memory and ended up saving 9GB of memory in total when VM template
+is enabled, which is about 72% of the total guest memory. See [full results
+here](https://github.com/kata-containers/runtime/pull/303#issuecomment-395846767).
+
+In another example, we creates 10 Kata containers with containerd shimv2
+and calculated the average boot up speed for each of them. And the result
+showed that VM template speeds up Kata containers creation by as much as
+38.68%. See [full results here](https://gist.github.com/bergwolf/06974a3c5981494a40e2c408681c085d).
+
+### What are the Cons
+One drawback of VM templating is that it cannot avoid cross-VM side-channel
+attack such as [CVE-2015-2877](https://access.redhat.com/security/cve/cve-2015-2877)
```

Comment:
I'm OK pointing to RH for the CVE - or we can point to one of the core sources such as https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-2877 - which are referenced off the RH item, so not a biggy.